app/models/placement_group.rb

```diff
@@ -0,0 +1,113 @@
+class PlacementGroup < ApplicationRecord
+  acts_as_miq_taggable
+  include YAMLImportExportMixin
+  include SupportsFeatureMixin
+  include NewWithTypeStiMixin
+  def self.new(*args, &block)
+    if self == Authentication && (args.empty? || args.first.kind_of?(Hash))
+      args = [{}] if args.empty?
+      h = args.first
+
+      type = h[inheritance_column.to_sym] || h[inheritance_column.to_s]
+      h[inheritance_column.to_sym] = "AuthUseridPassword" if type.nil?
+    end
+
+    super
+  end
+
+  belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "ManageIQ::Providers::CloudManager"
+  has_many :vms, :dependent => :nullify
+
+  delegate :queue_name_for_ems_operations, :to => :ext_management_system, :allow_nil => true
+
+  supports_not :safe_delete
+
+  def supports_safe_delete
+    supports?(:safe_delete)
+  end
+
+
+
+  def self.class_by_ems(ext_management_system)
+    ext_management_system&.class_by_ems(:CloudVolume)
+  end
```

Comment:
@Kuldip-Nanda it looks like you have a lot of duplication in your `ManageIQ::Providers::CloudManager::PlacementGroup` class and here in the base class.

There is nothing wrong with having that intermediate class but especially if these only exist in a CloudManager context it is likely extra complexity that you don't need.

Recommend moving the CRUD methods out of that and into this Operations module then deleting that class